### PR TITLE
Add plugin functionality

### PR DESCRIPTION
**Pull request by oyvindeide**

### CODE REVIEW

**Comment by berland** [comment]
More tests in https://github.com/equinor/semeio/tree/main/tests/jobs/stea to be copied over

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+[tool.setuptools_scm]
+write_to = "stea/version.py"
+
+[project.optional-dependencies]
+ert = ["ert"]
+test = [
+"fmu-steaclient[ert]",
+"pytest",
+"pytest-mock",
```
Comment (xjules) [inline pyproject.toml:48]: `pytest-httpserver`

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+exclude = '''
+
+(
+  /(
+      \.eggs         # exclude a few common directories in the
+    | \.git          # root of the project
+    | \.hg
+    | \.tox
+    | \.venv
+    | \stea/version.py
```
Comment (xjules) [inline pyproject.toml:62]: For some reasons this didn't get excluded. chatgpt suggested: `stea/version\.py`

**Review by berland — APPROVED** [review]
Approved!

**Pull request merged**